pkg/operator/controller/configurable-route/controller_test.go

```diff
@@ -0,0 +1,899 @@
+			componentRouteSpecs: []configv1.ComponentRouteSpec{
+				{
+					Name:                     "foo-route",
+					Namespace:                "foo-namespace",
+					ServingCertKeyPairSecret: configv1.SecretNameReference{},
+				},
+			},
+			componentRouteStatus: []configv1.ComponentRouteStatus{
+				{
+					Name:      "foo-route",
+					Namespace: "foo-namespace",
+					ConsumingUsers: []configv1.ConsumingUser{
+						"system:serviceaccount:foo:foo",
+					},
+				},
+			},
+			expectedAggregatedComponentRoute: []aggregatedComponentRoute{},
+		},
+		{
+			name: "invalid intersect due to different hashes",
+			componentRouteSpecs: []configv1.ComponentRouteSpec{
+				{
+					Name:      "foo-route",
+					Namespace: "bar-namespace",
+					ServingCertKeyPairSecret: configv1.SecretNameReference{
+						Name: "foo-secret",
+					},
+				},
+			},
+			componentRouteStatus: []configv1.ComponentRouteStatus{
+				{
+					Name:      "foo-route",
+					Namespace: "foo-namespace",
+					ConsumingUsers: []configv1.ConsumingUser{
+						"system:serviceaccount:foo:foo",
+					},
+				},
+			},
+			expectedAggregatedComponentRoute: []aggregatedComponentRoute{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actualAggregatedComponentRoute := intersectingComponentRoutes(tc.componentRouteSpecs, tc.componentRouteStatus)
+			assert.Equal(t, tc.expectedAggregatedComponentRoute, actualAggregatedComponentRoute)
+		})
+	}
+}
+
+// Test_deleteOrphanedRoles verifies deleteOrphanedRoles
+func Test_deleteOrphanedRoles(t *testing.T) {
+	testCases := []struct {
+		name           string
+		existingHashes sets.String
+		roles          rbacv1.RoleList
+		expectedRoles  rbacv1.RoleList
+		expectError    bool
+	}{
+		{
+			name: "all roles valid",
+			existingHashes: sets.String{
+				"foo-hash": sets.Empty{},
+			},
+			roles: rbacv1.RoleList{
+				Items: []rbacv1.Role{
+					*newRoleForServiceCert("foo-role", "openshift-config", "foo-hash", ""),
+				},
+			},
+			expectedRoles: rbacv1.RoleList{
+				Items: []rbacv1.Role{
+					*newRoleForServiceCert("foo-role", "openshift-config", "foo-hash", ""),
+				},
+			},
+		},
+		{
+			name: "foo-role doesn't have a hash",
+			existingHashes: sets.String{
+				"foo-hash": sets.Empty{},
+			},
+			roles: rbacv1.RoleList{
+				Items: []rbacv1.Role{
+					*newRoleForServiceCert("foo-role", "openshift-config", "", ""),
+				},
+			},
+			expectedRoles: rbacv1.RoleList{
+				Items: []rbacv1.Role{
+					*newRoleForServiceCert("foo-role", "openshift-config", "", ""),
+				},
+			},
+		},
+		{
+			name: "foo-role has outdated hash",
+			existingHashes: sets.String{
+				"foo-hash": sets.Empty{},
+			},
+			roles: rbacv1.RoleList{
+				Items: []rbacv1.Role{
+					*newRoleForServiceCert("foo-role", "openshift-config", "bar-hash", ""),
+				},
+			},
+			expectedRoles: rbacv1.RoleList{},
+		},
+		{
+			name:           "No existing hashes list",
```

Comment:
`go fmt` on this file?

Reply:
I use Goland and it continuously formats the code. Just to be sure, I ran `go fmt` and this is how it likes it.